modules/ROOT/pages/dataweave-type-selector.adoc

```diff
@@ -0,0 +1,209 @@
+= Type Selector
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+The type selector allows the use of the `.` selector over xref:dataweave-type-system.adoc#composite-types[types] which enables you to easily declare types from existing ones. This feature can also be combined with the new xref:dataweave-selectors-reusing-types.adoc[module loader feature] to load files and translate declarations inside your file into DataWeave type directives which can be accessed in the same way as types from any other module:
+
```

Comment:
== Example: Declaring a New Type from an Existing Type

The following example declares the `NameType` from the `User` type. Using the dot selector in `User.name` to declare the `name` variable a `NameType` enables the `typeOf()`  function to return the basic type, `"String"`. Notice how the DataWeave variable `name` is declared as a `NameType` value "Seba".

Reply:
Why does `typeOf()` return "String" and not "NameType"?

Reply:
Dev's input:
It has to do with how the typeOf() function works. It always returns the real primitive type (which is String). What I wanted to illustrate was how the String type was properly propagated from the NameType type. If you think this is confusing, we can shorten it to:
"
%dw 2.0
type User = {name: String}
type NameType = User.name
var name: NameType = "Seba"
"
and cut it there (no output), just to show how the selection works.